pygmt/clib/loading.py

```diff
@@ -37,85 +37,46 @@ def load_libgmt(env=None):
         couldn't access the functions).
 
     """
-    libpath = get_clib_path(env)
-    try:
-        libgmt = ctypes.CDLL(libpath)
-        check_libgmt(libgmt)
-    except OSError as err:
-        msg = "\n".join(
-            [
-                "Error loading the GMT shared library '{}':".format(libpath),
-                "{}".format(str(err)),
-            ]
-        )
-        raise GMTCLibNotFoundError(msg)
-    return libgmt
-
-
-def get_clib_path(env):
-    """
-    Get the path to the libgmt shared library.
-
-    Determine the file name and extension and append to the path set by
-    ``GMT_LIBRARY_PATH``, if any.
-
-    Parameters
-    ----------
-    env : dict or None
-        A dictionary containing the environment variables. If ``None``, will
-        default to ``os.environ``.
-
-    Returns
-    -------
-    libpath : str
-        The path to the libgmt shared library.
-
-    """
-    libname = clib_name()
     if env is None:
         env = os.environ
-    if "GMT_LIBRARY_PATH" in env:
-        libpath = os.path.join(env["GMT_LIBRARY_PATH"], libname)
-    else:
-        libpath = libname
-    return libpath
+    libnames = clib_name(os_name=sys.platform)
+    libpath = env.get("GMT_LIBRARY_PATH", "")
+    error = False
+    for libname in libnames:
+        try:
+            libgmt = ctypes.CDLL(os.path.join(libpath, libname))
+            check_libgmt(libgmt)
```

Comment:
If both ctypes.CDLL and check_libgmt don't raise any errors, it means we find the library, then we should break out of the loop. Right?
```suggestion
            check_libgmt(libgmt)
            break
```

Reply:
Of course! Thanks, Dongdong. I stopped coding yesterday because I couldn't even see this basic error anymore :)

Reply:
I just realized that this change is what's been causing our Windows builds to fail... Since we have a list of dll files to check, if one of the dll fails, the variable `error` turns `True` and stays `True` even if a later dll works. Our Linux and MacOS tests work because there's only one item in the list.